howto/stream/exchange-oob-data.md

```diff
@@ -24,83 +24,111 @@ The following instructions guide you to exchange OOB data using a specific imple
+          ...
+          dataChannels: {
+            "foo": {
+              callbacks: {
+                close: () => {
+                  console.log('data channel is closed')
+                },
+                open: () => {
+                  console.log('data channel is open')
+                },
+                error: (err) => {
+                  console.log(`error: ${err}`)
+                },
+                message: (data) => {
+                  console.log(`data received: ${data}`)
+                }
+              }
+            }
+          }
+        });
+        stream.connect();
+
+        const sendButton = document.getElementById("sendButton");
+        const textBox = document.getElementById("textBox");
+        sendButton.addEventListener("click", () => {
+          const message = textBox.value;
+          if (message) {
+            stream.sendData('foo', message);
+          }
+        });
+      };
+    </script>
+  </body>
+
 ```
 
+1. This extends a `AnboxStream` object to create a data channel named `foo` and registers event handlers for data communication between the Anbox instance and web client.
+2. It also adds a new section to the web page, containing an input field to type and send a message to Anbox runtime through the created data channel.
+
+
+
 ```{note}
 An `AnboxStream` object can create a maximum of five data channels. If the number of data channels exceeds the allowed maximum, an exception is thrown when instantiating the `AnboxStream` object.
 ```
 
-To launch a new WebRTC session, the client must call `stream.connect()`. The `AnboxStream` object then builds up a WebRTC native data channel internally, based on the name declared under its `dataChannels` property for peer-to-peer communication (`foo` in the example).
+### Data exchange between Anbox runtime and web client
 
-To send data to an Android application through the channel, the client must use the member function `sendData` of the `AnboxStream` class:
+Launch a stream-enabled instance
 
 ```
-stream.sendData('foo', 'hello world');
+amc launch --name test-app \
+  --raw jammy:android12:amd64 \
+  --enable-streaming
 ```
 
+Once the instance is up and running, retrieve the session id:
 
-### Anbox WebRTC platform
+    amc ls --filter name=test-app --format=csv | awk -F',' '{split($6, r, "="); print r[2]}'
 
-When establishing a peer connection, a number of data channels are set up in the Anbox WebRTC platform on the server side, upon request by the client.
+In the `service.py` file of the stream client sample project, replace `<session_id>` with the session ID retrieved above. Next, launch the web client by opening `https://<appliance_private_ip>:8080/<session_id>` in your browser. Once the web page is fully loaded, it establishes a WebRTC connection to the active session. After the WebRTC peer connection is successfully made, the `foo` data channel will be created on the Anbox runtime server side, in response to the client's request.
```

Comment:
We don't hard code the session id in the tutorial from https://documentation.ubuntu.com/anbox-cloud/en/latest/tutorial/stream-client/#implement-the-stream-client It is provided dynamically by the user including it in the URL he invokes.

```suggestion
In the `service.py` file of the stream client sample project, replace `<session_id>` with the session ID retrieved above. Next, launch the web client by opening `https://<appliance_private_ip>:8080/<session_id>` in your browser. Once the web page is fully loaded, it establishes a WebRTC connection to the instance. After the WebRTC peer connection is successfully made, the `foo` data channel will be created on the Anbox runtime server side, in response to the client's request.
```

"After the WebRTC peer connection" is also a new term we introduce here. Can we describe this in context of the `AnboxStream` object?

Reply:
Q: ...It is provided dynamically by the user including it in the URL he invokes.
A: ah yes, thanks for pointing this out

Q:"After the WebRTC peer connection" is also a new term we introduce here. Can we describe this in context of the AnboxStream object?
A: +1, yes we can.